src/plugin/message.cpp

```diff
@@ -141,7 +149,15 @@ namespace chapel {
     for (int i = 0; i < descriptor_->field_count(); i++) {
       printer->Print(vars[i],
         "when $field_number$ {\n");
-        if(vars[i]["proto_field_type"] == "enum") {
+        if(vars[i]["proto_field_type"] == "message") {
+          if(vars[i]["is_repeated"] == "0") {
+            printer->Print(vars[i],
+              "  $field_name$ = $proto_field_type$Consume(binCh, $type_name$);\n");
+          } else {
+            printer->Print(vars[i],
+              "  $field_name$.extend($proto_field_type$RepeatedConsume(binCh, $type_name$));\n");
+          }
+        } else if(vars[i]["proto_field_type"] == "enum") {
```

Comment:
Should the enum repeated case also be using `extend`, like this and the outermost else branch do?

Reply:
> Should the enum repeated case also be using `extend`, like this and the outermost else branch do?

We can use extend in `enums` as well. But a one-time assignment will also work, since the Consume function of `enum` is called only once(following the concept from the above comment) and returns encoded values of all fields together.

Reply:
Okay, that seems reasonable then :)